content/en/docs/upgrade/devops-data-migration-upgrade.md

```diff
@@ -0,0 +1,104 @@
+
+1. Retrieve information about DevOps directories from MySQL
+2. Retrieve information about pipelines and credentials from Jenkins
+3. Serialize the above information into a YAML file
+4. Back up the YAML resource object to the MinIO object storage
+5. Use the `kubectl apply` command to apply those pipeline resources
+6. Authorizations migration
+
+Data comparison before and after the migration:
+
+| Resource Affected |  v2.1   | v3.0 |
+| :---------------: | :-----: | :--: |
+|  DevOps projects  |  MySQL  | CRD  |
+|     Pipelines     | Jenkins | CRD  |
+|    Credentials    | Jenkins | CRD  |
+|  Authorizations   |  MySQL  | CRD  |
+
+{{< notice warning>}}
+
+You must back up the data in advance because the data in Jenkins will be overwritten by the ks-controller-manager once the CRD is created during data migration. You can use Velero to make a PVC-level backup, or back up the `/var/jenkins_home` directory of the Jenkins Pod directly.
+
+{{</ notice >}}
+
+## Data Migration Plans and Risks Evaluation
+
+For the pipelines created on the KubeSphere DevOps page, migration can be carried out as normal. However, the pipeline configurations directly made on the Jenkins dashboard might get lost during the migration.
+
+### Migration through a Job
+
+Follow the upgrade process as normal. This plan applys to the scenario where the DevOps operations are only made on the KubeSphere DevOps page. If there is no DevOps resources available after the upgrade, you can rerun the upgrade process.
+
+### No migration for the previous data
+
+KubeSphere DevOps is compatible with most Jenkins configurations, but not all of them. As shown in the below image, certain configurations on the Jenkins dashboard are not supported by KubeSphere DevOps.
+
+![configurations-not-supported](/images/docs/upgrade/devops-data-migration-upgrade/configurations-not-supported.png)
+
+Those configurations will be cleared after the data migration. Even if those configurations are added again after the data migration, they will still be cleared when the pipeline is edited next time.
+
+{{< notice note >}}
+
+Generally speaking, those configurations cannot be configured into the Jenkinsfile because they are not in the logic when running a pipeline.
+
+{{</ notice >}}
+
+If you have specific needs for those configurations, you can keep your previous data unmigrated and manage the new pipelines on the KubeSphere DevOps page. 
+
+{{< notice warning >}}
+
+Do not create pipelines with the same name under one DevOps project, or the data in Jenkins will be overwritten.
+
+{{</ notice >}}
+
+### Partial migration
+
+If there is only a small amount of configurations made on the Jenkins dashboard, users can make a backup first and then carry out a partial data migration without those special configurations by using a Job. Pipelines with special configurations can be restored later on, but it is recommended to manage pipelines on the KubeSphere DevOps page.
```

Comment:
->
If you only have few configurations to be specified on the Jenkins dashboard, you can use KubeSphere DevOps to manage pipelines directly. For these users, they can make a backup first and then migrate part of the data by using a Job while fixing pipelines with special configurations later.